lib/finch.ex

```diff
@@ -130,12 +130,20 @@ defmodule Finch do
     end
   end
 
-  defp build_method(method) when method in @methods, do: method
+  defp build_method(method) when is_binary(method), do: method
 
-  defp build_method(method) when is_atom(method) do
+  defp build_method(method) when is_atom(method) and method in @atom_methods do
     @atom_to_method[method]
```

Comment:
I think this can now become simply:

```suggestion
  defp build_method(method) when method in @atom_methods do
    method |> Atom.to_string() |> String.upcase()
```

and you can remove `@methods` and `@atom_to_method`.

Reply:
Ah right, I can definitely remove the is_atom/1 guard, but wouldnt the map lookup be slightly faster than the conversion?

Reply:
I did a quick benchmark and got these results:
```
Name                 ips        average  deviation         median         99th %
map access        8.81 M      113.53 ns   ±409.74%           0 ns        1000 ns
convert           3.44 M      290.47 ns  ±6416.53%           0 ns        1000 ns

Comparison:
map access        8.81 M
convert           3.44 M - 2.56x slower +176.95 ns
```